modelzoo/SimpleMultiTask/train.py

```diff
@@ -0,0 +1,790 @@
+                                               global_step=self.global_step)
+
+    # compute acc & auc
+    def _create_metrics(self):
+        self.acc, self.acc_op = tf.metrics.accuracy(labels=self.label,
+                                                    predictions=self.output)
+        self.auc, self.auc_op = tf.metrics.auc(labels=self.label,
+                                               predictions=self.probability,
+                                               num_thresholds=1000)
+        tf.summary.scalar('eval_acc', self.acc)
+        tf.summary.scalar('eval_auc', self.auc)
+
+    def _create_dense_layer(self, input, num_hidden_units, activation, layer_name):
+        with tf.variable_scope(layer_name, reuse=tf.AUTO_REUSE) as mlp_layer_scope:
+            dense_layer = tf.layers.dense(input,
+                                       units=num_hidden_units,
+                                       activation=activation,
+                                       name=mlp_layer_scope)
+            self._add_layer_summary(dense_layer, mlp_layer_scope.name)
+        return dense_layer
+
+    # create model
+    def _create_model(self):
+        with tf.variable_scope('input_layer',
+                                partitioner=self._input_layer_partitioner,
+                                reuse=tf.AUTO_REUSE):
+            if not self._tf and self._adaptive_emb:
+                '''Adaptive Embedding Feature Part 1 of 2'''
+                adaptive_mask_tensors = {}
+                for i in range(len(HASH_INPUTS)):
+                    adaptive_mask_tensors[HASH_INPUTS[i]] = tf.ones([self._batch_size],
+                                                                    tf.int32)
+                self._input_layer = tf.feature_column.input_layer(
+                        self.feature,
+                        self._feature_columns,
+                        adaptive_mask_tensors=adaptive_mask_tensors)
+            else:
+                self._input_layer = tf.feature_column.input_layer(self.feature, self._feature_columns)
+
+            if self._bf16:
+                self._input_layer = tf.cast(self._input_layer, dtype=tf.bfloat16)
+
+        with tf.variable_scope('SimpleMultiTask',
+                               reuse=tf.AUTO_REUSE):
+            self.clk_model, self.clk_logits = self._build_clk_model()
+            self.buy_model, self.buy_logits = self._build_buy_model()
+            self._logits = tf.squeeze(tf.stack([self.clk_model, self.buy_model], axis=1), [-1])
+            self.probability = self._logits
+            self.output = tf.round(self.probability)
+
+    def _build_clk_model(self):
+        if self._bf16:
+            d_clk = tf.cast(self._input_layer, dtype=tf.bfloat16)
+            with tf.variable_scope('clk_model',
+                                   partitioner=self._dense_layer_partitioner,
+                                   reuse=tf.AUTO_REUSE).keep_weights(dtype=tf.float32):
```

Comment:
Please use another way to use the BF16 feature in variable_scope. The _make_scope() you used in ESSM or like 
```
bst_scope = tf.variable_scope(
            'bst_tower',
            partitioner=self._dense_layer_partitioner,
            reuse=tf.AUTO_REUSE)
        with bst_scope.keep_weights() if self.bf16 else bst_scope:
```